### PR TITLE
reorganized codebase structure

### LINKED ISSUE #24: Enhancement: Commits and question description
Currently, it commits multiple times if a question has more than one tag, we should only create one commit per question.

### PR DESCRIPTION
**Pull request by haok1402**
resolves #24 #27

### CODE REVIEW

File: src/modules/github.ts [hunk 1]
```diff
@@ -0,0 +1,281 @@
...
+  if (!process.env.CLIENT_ID) {
+    await errorReport({
+      message: "CLIENT_ID not set",
+      context: "github.requestIdentity",
+    });
+  }
+
+  if (!process.env.CLIENT_SECRET) {
+    await errorReport({
+      message: "CLIENT_ID not set",
```
Comment (SenF1) [inline src/modules/github.ts]: Do you mean CLIENT_SECRET here?

File: src/modules/github.ts [hunk 2]
```diff
@@ -0,0 +1,281 @@
...
+      description: "Sync: LeetCode -> GitHub",
+      private: false,
+    });
+    while (!repository) {
+      await new Promise((resolve) => setTimeout(resolve, 2500));
+      const { data } = await octokit.rest.repos.listForUser({ username });
+      repository = data.find((repo) => repo.name === "LeetCode");
+    }
+    await notifyReport({
+      message: "Created a new GitHub repository for LeetCode!",
```
Comment (SenF1) [inline src/modules/github.ts:131]: I didn't receive any notification for new repo when testing, did this happen to you?

File: src/modules/leetcode.ts [hunk 3]
```diff
@@ -0,0 +1,240 @@
...
+        const suffix = extensionLookup[details.lang.name];
+        const payload: FileToCommit[] = [];
+        for (const tag of message.payload.topicTags) {
+          payload.push({
+            path: `${tag.slug}/${prefix}.${suffix}`,
+            content: message.payload.details.code,
+          });
+        }
+        const commitMessage = [
+          `:ballot_box_with_check: LC-${details.question.titleSlug}`,
```
Comment (SenF1) [inline src/modules/leetcode.ts:220]: Can we change back to brackets? Its easier to differentiate sections.

**Review by SenF1 — CHANGES_REQUESTED** [review]
Hmm, it seems like it introduced a new bug where (frequent?) commit will lead to "Uncaught (in promise) HttpError: Update is not a fast forward", which did not happen previously.

File: src/modules/github.ts (hunk 1 above)
Comment (haok1402) [reply src/modules/github.ts]: Nice Catch! Fix on the way

File: src/modules/leetcode.ts (hunk 3 above)
Comment (haok1402) [reply src/modules/leetcode.ts:220]: Fixed :)

**Comment by haok1402** [comment]
[Screencast from 03-24-2024 12:40:24 PM.webm](https://github.com/SyncLeet/extension/assets/89672451/60f3da9a-402f-46fd-8c6b-81d3c3d214db)

**Pull request closed without merging**